atomic_reactor/plugins/generate_sbom.py

```diff
@@ -203,47 +214,68 @@ def get_parent_images_nvr(self) -> List[Optional[str]]:
             parent_images_nvr.append(nvr)
         return parent_images_nvr
 
-    def get_parent_image_components(self, nvr: Optional[str]) -> List[Dict[str, Any]]:
-        parent_components = []
+    def get_parent_image_components(self, nvr: Optional[str]) -> Dict[Any, Any]:
+        parent_components = {}
 
         if not nvr:
             self.incompleteness_reasons.add('parent build is missing SBOM')
-            return []
+            return {}
 
         build = self.koji_session.getBuild(nvr)
 
         if self.check_build_state(build, nvr):
-            sbom_url = self.get_sbom_url_from_build(build)
-            if sbom_url:
-                parent_image_sbom_json = self.get_sbom_json(sbom_url)
-                parent_components = parent_image_sbom_json['components']
-
-                # add reasons from parent images
-                if 'incompleteness_reasons' in parent_image_sbom_json:
-                    for reason in parent_image_sbom_json['incompleteness_reasons']:
-                        self.incompleteness_reasons.add(reason.get('description'))
+            sbom_urls = self.get_sbom_urls_from_build(build)
+
+            if sbom_urls:
+                for platform in self.all_platforms:
+                    parent_image_sbom_json = self.get_sbom_json(sbom_urls[platform])
+                    parent_components[platform] = parent_image_sbom_json['components']
+
+                    # add reasons from parent images
+                    if 'incompleteness_reasons' in parent_image_sbom_json:
+                        for reason in parent_image_sbom_json['incompleteness_reasons']:
```

Comment:
Maybe
```
for reason in parent_image_sbom_json.get('incompleteness_reasons', {}):
```
would improve the readability, as we could omit the condition and one nested level

Reply:
changed